pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -110,17 +171,23 @@ public String getTaskType() {
   public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
     String taskType = MergeRollupTask.TASK_TYPE;
     List<PinotTaskConfig> pinotTaskConfigs = new ArrayList<>();
-
+    Set<String> candidateMergeTables = new HashSet<>();
     for (TableConfig tableConfig : tableConfigs) {
       if (!validate(tableConfig, taskType)) {
         continue;
       }
       String offlineTableName = tableConfig.getTableName();
       LOGGER.info("Start generating task configs for table: {} for task: {}", offlineTableName, taskType);
+      candidateMergeTables.add(offlineTableName);
 
       // Get all segment metadata
       List<SegmentZKMetadata> allSegments = _clusterInfoAccessor.getSegmentsZKMetadata(offlineTableName);
 
+      // Reset the watermark time if no segment found
+      if (allSegments.isEmpty()) {
```

Comment:
Same as `line 409`, for empty table we will not add the metric since the watermark is never set, so we won't need to remove the metric?

Reply:
As we discussed, we still want to keep this because we want to cover the case when we deleted the segments for the existing table. I added more comments.

Reply:
Move `resetDelayMetrics` to line 142 in the old code (within the `if (preSelectedSegments.isEmpty())` block.

Reply:
done